test/interval.jl

```diff
@@ -85,6 +85,7 @@ isinf(::TimeType) = false
         @test_throws DomainError convert(Int, Interval{Closed, Open}(10, 10))
         @test convert(Int, Interval{Closed, Closed}(10, 10)) == 10
         @test_throws DomainError convert(Int, Interval{Closed, Closed}(10, 11))
+        @test convert(Interval{Float64, Closed, Closed}, Interval(1,2)) == Interval{Float64, Closed, Closed}(1.0,2.0)
```

Comment:
Hmmm, `convert(Interval{Float64, Open, Open}, Interval(1,2))` isn't lossless. This means that the convert method should probably be defined as: `Base.convert(::Type{Interval{T}}, interval::Interval{S,L,R}) where {T,S,L,R}`

Reply:
Would it also be useful to include `Base.convert(::Type{Interval{T,L,R}}, interval::Interval{S}) where {T,S,L,R}` such that if you need to convert from Closed Closed to e.g. Open Open you can? Should this type of conversion include Closed -> Open but not Open -> Closed

Reply:
I think this would make more sense as a constructor; https://docs.julialang.org/en/v1/manual/conversion-and-promotion/#Conversion-vs.-Construction-1.

I'd probably leave this out unless you have a use case.